src/presenter/events-presenter.js

```diff
@@ -1,25 +1,51 @@
 import {render, replace} from '../framework/render.js';
-import EditEventView from '../view/edit-event-view.js';
-import EventView from '../view/event-view.js';
+import SortListView from '../view/sort-list-view.js';
 import EventsListView from '../view/events-list-view.js';
+import EventView from '../view/event-view.js';
+import EditEventView from '../view/edit-event-view.js';
+import MessageView from '../view/message-view.js';
+import {filtrate, DEFAULT_FILTER} from '../utils/filter.js';
 export default class EventsPresenter {
   #container = null;
   #eventsModel = null;
   #events = [];
   #destinations = [];
   #offers = [];
+  #isLoadFail = false;
+  #filter = DEFAULT_FILTER;
   #eventsListComponent = new EventsListView();
+  #sortListComponent = new SortListView();
 
   constructor({container, model}) {
     this.#container = container;
     this.#eventsModel = model;
   }
 
+  set filter(value) {
+    this.#filter = value;
+  }
+
+  get filter() {
+    return this.#filter;
+  }
+
   init() {
     this.#events = [...this.#eventsModel.events];
+    if (this.#filter !== DEFAULT_FILTER) {
+      this.#events = filtrate[this.#filter](this.#events);
+    }
+
     this.#destinations = [...this.#eventsModel.destinations];
     this.#offers = [...this.#eventsModel.offers];
 
+    if (this.#isLoadFail) {
+      render(new MessageView({errorLoading: true}), this.#container);
+      return;
+    } else if (!this.#isLoadFail && this.#events.length === 0) {
+      render(new MessageView({filter: this.#filter}), this.#container);
+      return;
+    }
```

Comment:
Здесь можно избавиться от лишнего `!this.#isLoadFail`, т.к. он проверяется в первой ветке `if`. И тогда ещё логично разделить это на 2 `if`, потому что это логически отдельные проверки, а не разные ветки одной логики.

```suggestion
if (this.#isLoadFail) {
  render(new MessageView({errorLoading: true}), this.#container);
  return;
}

if (!this.#events.length) {
  render(new MessageView({filter: this.#filter}), this.#container);
  return;
}
```